com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/MaterialBuilder.java

```diff
@@ -232,6 +232,17 @@ public Task<Void> create(IResource input) throws IOException, CompileExceptionEr
         task.addInput(vertexProgramOutputResource);
         task.addInput(fragmentProgramOutputResource);
 
+        for (MaterialDesc.Sampler materialSampler : materialBuilder.getSamplersList()) {
+            String texture = materialSampler.getTexture();
+            if (texture.isEmpty())
+                continue;
+            IResource res = BuilderUtil.checkResource(this.project, input, "texture", texture);
+            Task<?> embedTask = this.project.createTask(res);
+            if (embedTask == null) {
+                throw new CompileExceptionError(input, 0, String.format("Failed to create build task for component '%s'", res.getPath()));
+            }
+        }
```

Comment:
A new build dependency.
The sampler refers to a texture which needs to be compiled as well.